core/response_test.go

```diff
@@ -154,7 +154,7 @@ var _ = Describe("ResponseWriter tests", func() {
 			Expect(err).To(BeNil())
 
 			// Headers are not written to `Headers` field
```

Comment:
This comment makes it seem like this was intentional. If we want to use `Headers` and `MultiValueHeaders` then this comment does not make sense to keep.